diff --git a/src/AppBundle/Security/FormLoginAuthenticator.php b/src/AppBundle/Security/FormLoginAuthenticator.php
index 591611d..750a0d2 100644
--- a/src/AppBundle/Security/FormLoginAuthenticator.php
+++ b/src/AppBundle/Security/FormLoginAuthenticator.php
@@ -34,6 +34,7 @@ class FormLoginAuthenticator extends AbstractGuardAuthenticator
         }
 
         $username = $request->request->get('_username');
+        $request->getSession()->set(Security::LAST_USERNAME, $username);
         $password = $request->request->get('_password');
 
         return array(
