diff --git a/src/AppBundle/Security/FormLoginAuthenticator.php b/src/AppBundle/Security/FormLoginAuthenticator.php
index bd0f88c..b62dd19 100644
--- a/src/AppBundle/Security/FormLoginAuthenticator.php
+++ b/src/AppBundle/Security/FormLoginAuthenticator.php
@@ -72,7 +72,7 @@ class FormLoginAuthenticator extends AbstractGuardAuthenticator
 
     public function supportsRememberMe()
     {
-        // TODO: Implement supportsRememberMe() method.
+        return true;
     }
 
     public function start(Request $request, AuthenticationException $authException = null)
