diff --git a/src/AppBundle/Security/FormLoginAuthenticator.php b/src/AppBundle/Security/FormLoginAuthenticator.php
index d99d841..69a5b26 100644
--- a/src/AppBundle/Security/FormLoginAuthenticator.php
+++ b/src/AppBundle/Security/FormLoginAuthenticator.php
@@ -14,7 +14,17 @@ class FormLoginAuthenticator extends AbstractGuardAuthenticator
 {
     public function getCredentials(Request $request)
     {
-        // TODO: Implement getCredentials() method.
+        if ($request->getPathInfo() != '/login_check' || !$request->isMethod('POST')) {
+            return;
+        }
+
+        $username = $request->request->get('_username');
+        $password = $request->request->get('_password');
+
+        return array(
+            'username' => $username,
+            'password' => $password
+        );
     }
 
     public function getUser($credentials, UserProviderInterface $userProvider)
