diff --git a/src/AppBundle/Security/FormLoginAuthenticator.php b/src/AppBundle/Security/FormLoginAuthenticator.php
index 678f7a5..3cfd9e8 100644
--- a/src/AppBundle/Security/FormLoginAuthenticator.php
+++ b/src/AppBundle/Security/FormLoginAuthenticator.php
@@ -79,7 +79,7 @@ class FormLoginAuthenticator extends AbstractGuardAuthenticator
 
     public function supportsRememberMe()
     {
-        // TODO: Implement supportsRememberMe() method.
+        return true;
     }
 
     public function start(Request $request, AuthenticationException $authException = null)
