diff --git a/src/AppBundle/Security/FormLoginAuthenticator.php b/src/AppBundle/Security/FormLoginAuthenticator.php
index f88529f..bb2bed1 100644
--- a/src/AppBundle/Security/FormLoginAuthenticator.php
+++ b/src/AppBundle/Security/FormLoginAuthenticator.php
@@ -14,7 +14,10 @@ class FormLoginAuthenticator extends AbstractGuardAuthenticator
 {
     public function getCredentials(Request $request)
     {
-        // TODO: Implement getCredentials() method.
+        return array(
+            'username' => $request->request->get('_username'),
+            'password' => $request->request->get('_password')
+        );
     }
 
     public function getUser($credentials, UserProviderInterface $userProvider)
