diff --git a/src/AppBundle/Security/FormLoginAuthenticator.php b/src/AppBundle/Security/FormLoginAuthenticator.php
index d07a357..b76f296 100644
--- a/src/AppBundle/Security/FormLoginAuthenticator.php
+++ b/src/AppBundle/Security/FormLoginAuthenticator.php
@@ -3,8 +3,10 @@
 namespace AppBundle\Security;
 
 use KnpU\GuardBundle\Guard\AbstractGuardAuthenticator;
+use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
 use Symfony\Component\Security\Core\Exception\AuthenticationException;
@@ -16,9 +18,12 @@ class FormLoginAuthenticator extends AbstractGuardAuthenticator
 {
     private $encoder;
 
-    public function __construct(UserPasswordEncoderInterface $encoder)
+    private $router;
+
+    public function __construct(UserPasswordEncoderInterface $encoder, RouterInterface $router)
     {
         $this->encoder = $encoder;
+        $this->router = $router;
     }
 
     public function getCredentials(Request $request)
@@ -45,7 +50,9 @@ class FormLoginAuthenticator extends AbstractGuardAuthenticator
 
     public function onAuthenticationFailure(Request $request, AuthenticationException $exception)
     {
-        // TODO: Implement onAuthenticationFailure() method.
+        $url = $this->router->generate('security_login_form');
+
+        return new RedirectResponse($url);
     }
 
     public function onAuthenticationSuccess(Request $request, TokenInterface $token, $providerKey)
